src/routes/conversation/[id]/+server.ts

```diff
@@ -136,38 +138,53 @@ export async function POST({ request, locals, params, getClientAddress }) {
 		hashes = await Promise.all(files.map(async (file) => await uploadFile(file, conv)));
 	}
 
+	// can only call isContinue on the last message id
+	if (
+		isContinue &&
+		conv.messages.findIndex((message) => message.id === messageId) !== conv.messages.length - 1
+	) {
+		throw error(400, "Can only continue the last message");
+	}
+
 	// get the list of messages
 	// while checking for retries
 	let messages = (() => {
-		if (is_retry && messageId) {
+		// for retries we slice and rewrite the last user message
+		if (isRetry && messageId) {
 			// if the message is a retry, replace the message and remove the messages after it
 			let retryMessageIdx = conv.messages.findIndex((message) => message.id === messageId);
+
 			if (retryMessageIdx === -1) {
 				retryMessageIdx = conv.messages.length;
 			}
+
 			return [
 				...conv.messages.slice(0, retryMessageIdx),
 				{
-					content: newPrompt,
+					content: conv.messages[retryMessageIdx]?.content,
 					from: "user",
 					id: messageId as Message["id"],
 					updatedAt: new Date(),
 					files: conv.messages[retryMessageIdx]?.files,
 				},
 			];
+		} else if (isContinue && messageId) {
+			// for continue we do nothing and expand the last assistant message
+			return [...conv.messages];
```

Comment:
```suggestion
			return conv.messages;
```

why not just this ? do we need to clone it [... xyz] ?

Reply:
fixed it!